cli/src/pcluster/utils.py

```diff
@@ -243,6 +243,10 @@ def policy_name_to_arn(policy_name):
     return "arn:{0}:iam::aws:policy/{1}".format(get_partition(), policy_name)
 
 
+def get_resource_name_from_resource_arn(resource_arn):
```

Comment:
let's create unit test for it.

You might also create a function to verify if an arn is an instance-profile given that you have the split and check for "instance-profile" in two places.

Then given that it's an utility related to "AWS" I'd move it in the src/pcluster/aws module.
We can create a new src/pcluster/aws/aws_utils.py there to group all the AWS related functions.

Reply:
There are already unit test for this function in test_iam_validators

Since we now separate InstanceRole and InstanceProfile, we don't need to function to verify if an arn is an instance-profile